src/Analyzer/Pass/Expression/ArrayDuplicateKeys.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * @author Kévin Gomez https://github.com/K-Phoen <contact@kevingomez.fr>
+ */
+
+namespace PHPSA\Analyzer\Pass\Expression;
+
+use PhpParser\Node\Expr;
+use PhpParser\Node\Scalar;
+use PHPSA\Analyzer\Pass;
+use PHPSA\Context;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+
+class ArrayDuplicateKeys implements Pass\AnalyzerPassInterface, Pass\ConfigurablePassInterface
+{
+    /**
+     * @param Expr\Array_ $expr
+     * @param Context $context
+     * @return bool
+     */
+    public function pass(Expr\Array_ $expr, Context $context)
+    {
+        $result = false;
+        $keys = [];
+
+        /** @var Expr\ArrayItem $item */
+        foreach ($expr->items as $item) {
+            if (!$item->key instanceof Scalar) {
```

Comment:
I think what he meant was to compile the $expr variable. This will create an Array CompiledExpression and all items will be CompiledExpressions too (key and value). This should compile variable keys down to their value ... i think
So if we have 
`$a = 10;
$array = [10 => 1, $a => 2];`
with CompiledExpression we would get a key collision whereas with your approach i think not? Because your keys should be "10" and "PHPSA_$a".